docs/src/pages/documentation/cloudflare-workers.md

```diff
@@ -0,0 +1,375 @@
+---
+title: "Deploying Web Fragments on Cloudflare Workers"
+layout: "~/layouts/MarkdownLayout.astro"
+---
+
+_Last updated_: October 28, 2025
+
+Cloudflare Workers provides an excellent platform for deploying Web Fragments, offering global edge deployment and powerful service integration features. This guide walks you through deploying a Web Fragments architecture on Cloudflare, with special attention to patterns that maximize performance and maintainability.
+
+## Understanding the Cloudflare Workers Architecture
+
+When deploying Web Fragments on Cloudflare, you'll typically work with multiple workers that communicate using **Service Bindings** rather than traditional HTTP requests. Think of Service Bindings as direct, internal connections between your workers; like having a private network where your services can talk to each other without going through the public internet.
+
+Here's the typical setup:
+
+```
+User's Browser
+     ↓
+Gateway Worker (public-facing)
+     ├─→ Fragment Worker 1 (via Service Binding)
+     ├─→ Fragment Worker 2 (via Service Binding)
+     └─→ Static Assets (via Assets Binding)
+```
+
+This architecture gives you several advantages:
+
+- **Performance**: Service Bindings eliminate network overhead
+- **Security**: Backend workers don't need public URLs
+- **Simplicity**: No CORS configuration needed
+- **Cost**: No outbound HTTP requests between your own workers
+
+## Setting Up Your Project
+
+### 1. Create a Monorepo Structure (optional)
+
+We recommend organizing your project as a monorepo using pnpm workspaces. This makes it easier to manage multiple workers and share dependencies:
+
+```
+my-app/
+├── pnpm-workspace.yaml
+└── packages/
+    ├── gateway/          # Public-facing gateway worker
+    ├── shell-app/        # Your existing SPA (React, Vue, etc.)
+    └── fragment-app/     # Your web fragment
+```
+
+Create a `pnpm-workspace.yaml` at the root:
+
+```yaml
+packages:
+  - "packages/*"
+```
+
+### 2. Configure Your Fragment Worker
+
+Let's start with the fragment. The independent application you want to embed. Here's a typical `wrangler.toml` for a Remix-based fragment:
+
+```toml
+# packages/fragment-app/wrangler.toml
+name = "my-fragment"
+main = "./workers/app.ts"
+compatibility_date = "2025-10-28"
+
+# This worker doesn't need a public URL
+# It will only be accessed via Service Binding
+workers_dev = false
+```
```

Comment:
and please turn on logs, workers observability, smart placement with a hint, and most importantly assets